pkg/controllers/user/alert/watcher/workload.go

```diff
@@ -162,11 +181,63 @@ func (w *WorkloadWatcher) checkWorkloadCondition(wl *workload.Workload, alert *v
 		data["workload_kind"] = wl.Kind
 		data["available_percentage"] = strconv.Itoa(percentage)
 		data["available_replicas"] = strconv.Itoa(int(wl.Status.AvailableReplicas))
-		data["desired_replicas"] = strconv.Itoa(int(wl.Status.Replicas))
+		data["desired_replicas"] = strconv.Itoa(int(desiredReplicas))
 
 		if err := w.alertManager.SendAlert(data); err != nil {
 			logrus.Errorf("Failed to send alert: %v", err)
 		}
 	}
 
 }
+
+func (w *WorkloadWatcher) getDesiredReplicas(workloadName string) (int32, error) {
+	var desiredReplicas int32
+	splitted := strings.Split(workloadName, ":")
+	if len(splitted) != 3 {
+		return desiredReplicas, errors.New("invalid workloadName: " + workloadName)
+	}
+	workloadType := strings.ToLower(splitted[0])
+	namespace := splitted[1]
+	name := splitted[2]
+	switch workloadType {
+	case workload.ReplicationControllerType:
```

Comment:
we already workloadController https://github.com/rancher/rancher/pull/21638/files#diff-2046287b69957361f7d174a68bcc0d66R62, could we use it to GetWorkload?

Reply:
Hi @kinarashah workload struct isn't include desired replicas, just include current replicas, I afraid change workload struct affect other parts, so I get desired replicas here.

Reply:
Unless I'm missing something, we have both desired and available replicas? 
https://github.com/rancher/rancher/blob/1a7a0b5329cabcff30fda8d5a75108a779846605//pkg/controllers/user/workload/workload_common.go#L262 

https://github.com/rancher/rancher/blob/1a7a0b5329cabcff30fda8d5a75108a779846605//pkg/controllers/user/workload/workload_common.go#L340

Reply:
Hi @kinarashah , only the daemonSet use desired number, statefulSet, deployment, replicaSet use status replicas, it is the most recently observed number of replicas, so I finally give up function in workload_common.